src/server/proxy_protocol.rs

```diff
@@ -1,15 +1,18 @@
-use super::chancomms::ProxyLoopSender;
-use super::session::SharedSession;
-use crate::server::chancomms::ProxyLoopMsg;
+use super::{
+    chancomms::{ProxyLoopMsg, ProxyLoopSender},
+    session::SharedSession,
+};
 use crate::{auth::UserDetail, storage::StorageBackend};
 use bytes::Bytes;
-use proxy_protocol::{parse, version1::ProxyAddresses, ProxyHeader};
-use std::net::{SocketAddr, SocketAddrV4};
-use std::{collections::HashMap, net::IpAddr, ops::Range};
+use dashmap::{mapref::entry::Entry, DashMap};
+use proxy_protocol::{parse, version1::ProxyAddresses, ParseError, ProxyHeader};
+use std::net::{IpAddr, SocketAddr, SocketAddrV4};
+use std::ops::Range;
+use thiserror::Error;
 use tokio::io::AsyncReadExt;
 
 #[derive(Clone, Copy, Debug)]
-pub enum ProxyMode {
+pub(super) enum ProxyMode {
```

Comment:
What necessitates reducing the scope to super?

Reply:
It's not necessary because it's not exported but it's just to clarify the scope. If you think it's not common practice to do it this way, I'll revert it.

Reply:
I’m not sure if its common tbh. I don’t mind it, was just wondering.